include/alpaka/core/BindScope.hpp

```diff
@@ -0,0 +1,244 @@
+/* Copyright 2019 Benjamin Worpitz, Matthias Werner
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+#include <alpaka/core/Assert.hpp>
+
+#include <type_traits>
+
+// ALPAKA_LAMBDA defines compiler dependent host-device annotation and capture
+// - from: https://github.com/kokkos/kokkos/wiki/Lambda-Dispatch
+#if BOOST_COMP_NVCC || BOOST_COMP_CLANG_CUDA
+
+#if BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(8, 0, 0)
+  #define ALPAKA_LAMBDA [=]__device__
+#else
+  #define ALPAKA_LAMBDA [=]__host__ __device__
+  #if ALPAKA_CXX_STANDARD>14
+    #define ALPAKA_CLASS_LAMBDA [=,*this] __host__ __device__
+  #endif
+#endif
+
+#else // HIP(HCC) and CPU backends
+  // lambdas do not require a host-device attributes
+  #define ALPAKA_LAMBDA [=]
```

Comment:
Could we remove the lambda capture from the macro and create a `ALPAKA_FN_LAMBDA` equivalent to `ALPAKA_FN_HOST_ACC` which only contains the attributes?

Reply:
for heterogeneous lambda kernels copy-by-reference is not really feasible, so always doing "[=]" + relying on compilers ability to only copy used variables was probably the idea, when kokkos introduced this macro. At least our lambdas are not only kernels and Alpaka is highly explicit anyways. So yes, we could leave it to the user in this current version.

Reply:
After thinking about it I would also prefer `ALPAKA_FN_LAMBDA` or 

```
#define ALPAKA_LAMBDA_CAPTURE( ... ) [=, __VA_ARGS__ ] __host__ __device__
```